crux-llvm/src/Crux/LLVM/Overrides.hs

```diff
@@ -270,6 +277,44 @@ lib_fresh_i64 ::
   GlobalVar Mem -> Fun sym (LLVM arch) (EmptyCtx ::> TPtr arch) (TBits 64)
 lib_fresh_i64 mem = lib_fresh_bits mem knownNat
 
+lib_fresh_str ::
+  (ArchOk arch, IsSymInterface sym) =>
+  GlobalVar Mem -> Fun sym (LLVM arch) (EmptyCtx ::> TPtr arch ::> TBits (ArchWidth arch)) (TPtr arch)
+lib_fresh_str mvar = do
+  RegMap args <- getOverrideArgs
+  case args of
+    Empty :> pName :> maxLen -> do
+      name <- lookupString mvar pName
+      sym <- getSymInterface
+
+      -- Compute the allocation length, which is the requested length plus one
+      -- to hold the NUL terminator
+      one <- liftIO $ bvLit sym ?ptrWidth 1
+      maxLenBV <- liftIO $ projectLLVM_bv sym (regValue maxLen)
+      len <- liftIO $ bvAdd sym maxLenBV one
+      mem0 <- readGlobal mvar
+
+      -- Allocate memory to hold the string
+      (ptr, mem1) <- liftIO $ doMalloc sym HeapAlloc Immutable name mem0 len noAlignment
+
+      -- Allocate contents for the string - we want to make each byte symbolic,
+      -- so we allocate a fresh array (which has unbounded length) with symbolic
+      -- contents and write it into our allocation.  This write does not cover
+      -- the NUL terminator.
+      contentsName <- case userSymbol (name ++ "_contents") of
+        Left err -> fail (show err)
+        Right nm -> return nm
+      let arrayRep = BaseArrayRepr (Empty :> BaseBVRepr ?ptrWidth) (BaseBVRepr (knownNat @8))
+      initContents <- liftIO $ freshConstant sym contentsName arrayRep
+      mem2 <- liftIO $ doArrayConstStore sym mem1 ptr noAlignment initContents maxLenBV
+
+      -- Write the NUL terminator to the last byte
+      zeroByte <- liftIO $ bvLit sym (knownNat @8) 0
```

Comment:
`arrayUpdate`  is an alternative way to write the final `0` in the array, and avoid a second write